File: src/main/java/org/hotel/hotelreservationsystemydg/service/ReservationService.java
package org.hotel.hotelreservationsystemydg.service;

import org.hotel.hotelreservationsystemydg.dto.CheckInRequestDto;
import org.hotel.hotelreservationsystemydg.dto.CheckOutRequestDto;
import org.hotel.hotelreservationsystemydg.dto.ReservationDateRangeDto;
import org.hotel.hotelreservationsystemydg.dto.ReservationRequestDto;
import org.hotel.hotelreservationsystemydg.dto.ReservationResponseDto;

import java.util.List;

public interface ReservationService {

    ReservationResponseDto createReservation(ReservationRequestDto request);

    void checkIn(CheckInRequestDto request);

    void checkOut(CheckOutRequestDto request);

    List<ReservationResponseDto> getReservations();

    List<ReservationDateRangeDto> getReservationsByRoomId(Long roomId);

    ReservationResponseDto getReservationByCode(String reservationCode);

    ReservationResponseDto cancelReservation(Long reservationId);

}

File: frontend/src/services/reservationService.js
import api from "./api";

export const createReservation = (payload) => {
  return api.post("/reservations/createReservation", payload);
};

export const getReservationsByRoomId = (roomId) => {
  return api.get(`/reservations/room/${roomId}`);
};

export const getReservationByCode = (reservationCode) => {
  return api.get(`/reservations/code/${reservationCode}`);
};
